### ISSUE TITLE
Error when selecting gene expression using interactive viewer

### ISSUE THREAD

**Issue by MarionPerrier** [theislab/squidpy#485]
## Description

Hi !

I have an error when trying to view gene expression with the graphical interface, while following the tutorial. This cause python to shut down.
...

## Minimal reproducible example

```python
Python 3.8.12 | packaged by conda-forge | (default, Jan 30 2022, 23:53:36) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
import scanpy as sc
import squidpy as sq
import numpy as np
adata = sq.datasets.visium_hne_adata()
img = sq.datasets.visium_hne_image()
viewer = img.interactive(adata)
...
```

## Traceback

After clicking on a gene on the right panel :
<details>

```pytb
>>> /home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/numpy/core/numeric.py:2446: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return bool(asarray(a1 == a2).all())
/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py:297: FutureWarning: Public access to Window.qt_viewer is deprecated and will be removed in
v0.5.0. It is considered an "implementation detail" of the napari
application, not part of the napari viewer model. If your use case
requires access to qt_viewer, please open an issue to discuss.
  points_controls = self.view.viewer.window.qt_viewer.controls.widgets[layer]
WARNING: Traceback (most recent call last):
  File "/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_widgets.py", line 41, in <lambda>
    self.itemDoubleClicked.connect(lambda item: self._onAction((item.text(),)))
  File "/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_widgets.py", line 146, in _onAction
    self._controller.add_points(vec, key=item, layer_name=name)
  File "/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py", line 181, in add_points
    self._hide_points_controls(layer, is_categorical=is_categorical_dtype(vec))
  File "/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py", line 330, in _hide_points_controls
    gl.replaceWidget(labels[label_key], QLabel("percentile:"))
KeyError: 'n-dim:'
Abandon (core dumped)
...
```
Any idea on what could cause that ?

Thank you a lot for your time.
</details>

## Version

scanpy==1.8.2 anndata==0.8.0 umap==0.5.2 numpy==1.21.5 scipy==1.8.0 pandas==1.4.1 scikit-learn==1.0.2 statsmodels==0.13.2 python-igraph==0.9.9 pynndescent==0.5.6

squidpy==1.1.2
...

**Comment by giovp** [comment]
hi @MarionPerrier 

thanks for your interest in Squidpy. Did you try out in a ipython session? Could you try out in a jupyter notebook/lab session?

**Comment by MarionPerrier** [comment]
Thank you for your response ! 

No, I used python directly.
I have given a try with ipython, and python didn't shut down this time, but I have this result : 
![black_dots](https://user-images.githubusercontent.com/43764623/158987065-572f2519-bf4a-49be-9653-53b5a226012c.png)

And still the message with "KeyError: 'n-dim'"

```pytb
In [3]: viewer = img.interactive(adata)

/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/numpy/core/numeric.py:2446: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return bool(asarray(a1 == a2).all())
/home/user/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py:297: FutureWarning: Public access to Window.qt_viewer is deprecated and will be removed in
v0.5.0. It is considered an "implementation detail" of the napari
application, not part of the napari viewer model. If your use case
requires access to qt_viewer, please open an issue to discuss.
  points_controls = self.view.viewer.window.qt_viewer.controls.widgets[layer]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File ~/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_widgets.py:41, in ListWidget.__init__.<locals>.<lambda>(item)
     38 self._unique = unique
     39 self._controller = controller
---> 41 self.itemDoubleClicked.connect(lambda item: self._onAction((item.text(),)))
     42 self.enterPressed.connect(self._onAction)
     43 self.indexChanged.connect(self._onAction)

File ~/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_widgets.py:146, in AListWidget._onAction(self, items)
    144     logg.error(e)
    145     continue
--> 146 self._controller.add_points(vec, key=item, layer_name=name)

File ~/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py:181, in ImageController.add_points(self, vec, layer_name, key)
    168 layer: Points = self.view.viewer.add_points(
    169     self.model.coordinates,
    170     name=layer_name,
   (...)
    178     **properties,
    179 )
    180 # TODO(michalk8): add contrasting fg/bg color once https://github.com/napari/napari/issues/2019 is done
--> 181 self._hide_points_controls(layer, is_categorical=is_categorical_dtype(vec))
    182 layer.editable = False
    184 return True

File ~/miniconda3/envs/napari-env/lib/python3.8/site-packages/squidpy/pl/_interactive/_controller.py:330, in ImageController._hide_points_controls(self, layer, is_categorical)
    324 slider = RangeSlider(
    325     layer=layer,
    326     colorbar=self.view._colorbar,
    327 )
    328 slider.valueChanged.emit((0, 100))
--> 330 gl.replaceWidget(labels[label_key], QLabel("percentile:"))
    331 gl.replaceWidget(widget, slider)

KeyError: 'n-dim:'
```

Those black points are here for everything I try to visualize (clusters, genes expression, etc). Any idea on what might cause that ? 

This is a screenshot of the window : 
![screen_squidpy](https://user-images.githubusercontent.com/43764623/158987798-b51190ab-4356-4a62-91a9-19af1d71a9e5.png)

Thank you a lot for your time.

Marion

**Comment by michalk8** [comment]
Hi @MarionPerrier ,
You'd need to install the newest version that's on master as `pip install git+https://github.com/theislab/squidpy@master` and `napari==0.4.14`.
The `KeyError: n-dim:` is was fixed in our codebase in #453 , but the black dots are new to me (can confirm the occur in `napari==0.4.15`, which came 8 days ago).

**Comment by MarionPerrier** [comment]
I just tested, and the new SquidPy version combined with napari 0.4.14 works very fine, even for the dots (that aren't black anymore).

Thank you a lot for your help !
Have a great day.
